src/main/java/htsjdk/samtools/QueryInterval.java

```diff
@@ -42,10 +42,10 @@ public int compareTo(final QueryInterval other) {
     }
 
     /**
-     * @return true if both are on same reference, and other starts exactly where this ends.
+     * @return true if both are on same reference, and other starts exactly before this ends.
```

Comment:
I don't think we should be changing the definition of "abuts". since we use closed intervals the implementation was correct before the change.

Reply:
@yfarjoun If `QueryInterval` really uses closed intervals, then I think @Kirvolque is right and the existing implementation was wrong.  E.g. two closed intervals `1-9` and `10-20` abut one another,  but the coordinates are indeed off by one.

That said, I think this method should be private since it doesn't test bi-directionally.  That's fine given how it's used in `optimizeOverlaps`, but it would be confusing if called by other code.

Reply:
Ahh, I see. Sorry to judge so quick. The method is called from BAMFileReader, so it can't be changed to private. perhaps we just change it's name to something like `endAtStartOf` so that the code would read `prev.endsAtStartOf(nextInterval)`.

Reply:
@yfarjoun, thank you. The method has been renamed.